### PR TITLE
[Universal Profiling] Add inputs for collector/symbolizer

### PR DESCRIPTION
**Pull request by inge4pres**
## What does this PR do?

Add configuration options for universal_profiling integrations (collector/symbolizer) to set logging, telemetry and memory limits for the running binaries.

## Checklist

- [X] I have reviewed [tips for building integrations](https://github.com/elastic/integrations/blob/main/docs/tips_for_building_integrations.md) and this pull request is aligned with them.
- [X] I have verified that all data streams collect metrics or logs.
- [x] I have added an entry to my package's `changelog.yml` file.
- [X] I have verified that Kibana version constraints are current according to [guidelines](https://github.com/elastic/elastic-package/blob/master/docs/howto/stack_version_support.md#when-to-update-the-condition).

## Author's Checklist

- [ ] test the new options are propagated through Fleet

## How to test this PR locally

* Run a local stack with `elastic-stack up` (stack version 8.10 or higher)
* Build an install the new packages in Kibana
* Create an agent policy and enroll an agent locally
* Configure the new `vars`
* Validate they are actually used in the binaries

### CODE REVIEW

**Comment by inge4pres** [comment]
@elastic/fleet there's a CI failure trying to fetch Kibana image

```
docker.elastic.co/kibana/kibana:8.10.0
```

is it expected?

File: packages/universal_profiling_collector/changelog.yml [hunk 1]
```diff
@@ -1,4 +1,9 @@
 # newer versions go on top
+- version: 8.10.0
+  changes:
+    - description: Add input configuration options for day2 operations
```
Comment (florianl) [inline packages/universal_profiling_collector/changelog.yml:4]:
```suggestion
    - description: Add input configuration options
```

File: packages/universal_profiling_collector/manifest.yml [hunk 2]
```diff
@@ -25,9 +25,26 @@ policy_templates:
...
+          - name: telemetry
+            type: bool
+            default: true
+            description: Enable metrics
+          - name: memory_limit
+            type: text
+            default: ""
+            description: |
+              Set a soft memory limit for the program. The format for Megabytes is "MB", for Mebibytes is "MiB", etc...
+              The default value ("") does not apply any limits
```
Comment (florianl) [inline packages/universal_profiling_collector/manifest.yml:42]: `[..] does not apply any limits` is not correct.
```suggestion
              The default value ("") applies a memory limit of 25% of the available memory.
```

File: packages/universal_profiling_symbolizer/changelog.yml [hunk 3]
```diff
@@ -1,4 +1,10 @@
 # newer versions go on top
+- version: 8.10.0
+  changes:
+    - description: Add input configuration options for day2 operations
```
Comment (florianl) [inline packages/universal_profiling_symbolizer/changelog.yml:4]:
```suggestion
    - description: Add input configuration options
```

File: packages/universal_profiling_symbolizer/manifest.yml [hunk 4]
```diff
@@ -25,6 +25,27 @@ policy_templates:
...
+          - name: telemetry
+            type: bool
+            default: true
+            description: Enable metrics
+          - name: memory_limit
+            type: text
+            default: ""
+            description: |
+              Set a soft memory limit for the program. The format for Megabytes is "MB", for Mebibytes is "MiB", etc...
+              The default value ("") does not apply any limits
```
Comment (florianl) [inline packages/universal_profiling_symbolizer/manifest.yml:42]: `[..] does not apply any limits` is not correct.

```suggestion
              The default value ("") applies a memory limit of 25% of the available memory.
```

File: packages/universal_profiling_symbolizer/manifest.yml [hunk 5]
```diff
@@ -25,6 +25,27 @@ policy_templates:
...
+          - name: endpoint
+            type: text
+            default: ""
+            description: |
+              Set the endpoint for the object storage used to fetch public OS debug symbols; do not set unless you know
+              what you are doing
```
Comment (florianl) [inline packages/universal_profiling_symbolizer/manifest.yml:48]: I'm not the best in describing something, but as this is user facing, I think it should be reworded.

File: packages/universal_profiling_symbolizer/manifest.yml [hunk 6]
```diff
@@ -25,6 +25,27 @@ policy_templates:
...
+            description: Enable debug logging
+          - name: telemetry
+            type: bool
+            default: true
+            description: Enable metrics
+          - name: memory_limit
+            type: text
+            default: ""
+            description: |
+              Set a soft memory limit for the program. The format for Megabytes is "MB", for Mebibytes is "MiB", etc...
```
Comment (rockdaboot) [inline packages/universal_profiling_symbolizer/manifest.yml:41]:
```suggestion
              Specify a maximum memory allowance for the program. Memory units like 'MB' (Megabytes) or 'MiB' (Mebibytes) are supported.
```

**Comment by rockdaboot** [comment]
> @elastic/fleet there's a CI failure trying to fetch Kibana image
> 
> ```
> docker.elastic.co/kibana/kibana:8.10.0
> ```
> 
> is it expected?

I can only pull `8.10.1` or `8.10.2`.

**Comment by inge4pres** [comment]
> I can only pull `8.10.1` or `8.10.2`.

Thanks for the hint, I _think_ we then can solve it by increasing the constraint Kibana version to 8.10.2

File: packages/universal_profiling_collector/manifest.yml (hunk 2 above)
Comment (inge4pres) [reply packages/universal_profiling_collector/manifest.yml:42]: good pint, thanks 👍🏼

File: packages/universal_profiling_symbolizer/manifest.yml (hunk 5 above)
Comment (inge4pres) [reply packages/universal_profiling_symbolizer/manifest.yml:48]: WDYT of

```suggestion
              Set the endpoint for the object storage used to fetch public OS debug symbols.
              Change it only after understanding its effects,: a wrong endpoint will degrade the product.
```

Comment (rockdaboot) [reply packages/universal_profiling_symbolizer/manifest.yml:48]: "fetch" -> "retrieve" ?
And another suggestion for the second sentence:
"Only make changes if you fully understand the consequences, as an incorrect endpoint could negatively impact the product's performance."

Comment (danielmitterdorfer) [reply packages/universal_profiling_symbolizer/manifest.yml:48]: I glanced at https://www.elastic.co/guide/en/elasticsearch/reference/8.10/index-modules-store.html, which also describes a rather dangerous setting and came up with this:

> This is an expert-only setting. An incorrect endpoint could negatively impact the product's performance. We recommend sticking to the defaults.

File: packages/universal_profiling_collector/manifest.yml [hunk 7]
```diff
@@ -25,9 +25,26 @@ policy_templates:
...
+            description: Enable debug logging
+          - name: telemetry
+            type: bool
+            default: true
+            description: Enable metrics
+          - name: memory_limit
+            type: text
+            default: ""
+            description: |
+              Set a soft memory limit for the program. The format for Megabytes is "MB", for Mebibytes is "MiB", etc...
```
Comment (rockdaboot) [inline packages/universal_profiling_collector/manifest.yml:41]: Why not use the same description as in the symbolizer's manifest ?

**Review by florianl — APPROVED** [review]
works fine :+1: we can adjust descriptions later on as well.

File: packages/universal_profiling_collector/manifest.yml (hunk 7 above)
Comment (inge4pres) [reply packages/universal_profiling_collector/manifest.yml]: Updated in https://github.com/elastic/integrations/pull/7962/commits/2610a386ec3af7902a8373f18e36efeefd1fbf7f

**Pull request merged**